server/server.go

```diff
@@ -260,7 +260,9 @@ func (s *server) Endpoints() []Endpoint {
 }
 
 func (s *server) ErrorEncoder() kithttp.ErrorEncoder {
-	return func(ctx context.Context, err error, w http.ResponseWriter) {
+	return func(ctx context.Context, serverError error, w http.ResponseWriter) {
+		var err error
```

Comment:
We had an issue with the `err` variable. Different errors overwrote the original one which we used to enrich the error logging etc.. E.g. the error trace was not parsed correctly.